demo.py

```diff
@@ -0,0 +1,119 @@
+    update.message.reply_text(text=text, reply_markup=keyboard)
+    return DemoDecisions.AMAZON
+
+
+def demo_durable(update, context):
+    text = "Mohammed Ali or Covid-19, who's more durable?"
+    keyboard = ReplyKeyboardMarkup([["Mohammed Ali", "Covid-19"]])
+    update.message.reply_text(text=text, reply_markup=keyboard)
+    return DemoDecisions.DURABLE
+
+
+def demo_difficult(update, context):
+    text = "What's more difficult do build?"
+    keyboard = ReplyKeyboardMarkup([["Chinese Wall", "Golden-Gate-Bridge"], ["Eiffel Tower", "Colosseum", "Covid-19 vaccine"]])
+    update.message.reply_text(text=text, reply_markup=keyboard)
+    return DemoDecisions.DIFFICULT
+
+
+def demo_michael(update, context):
+    text = "If Michael was alive, would he know how to heal Covid-19?"
+    keyboard = ReplyKeyboardMarkup([["Yes", "No", "Not so sure"]])
+    update.message.reply_text(text=text, reply_markup=keyboard)
+    return DemoDecisions.MICHAEL
+
+
+def demo_winner(update, context):
+    text = "Who wins eventually?"
+    keyboard = ReplyKeyboardMarkup([["Covid-19", "Humanity"]])
+    update.message.reply_text(text=text, reply_markup=keyboard)
+    return DemoDecisions.WINNER
+
+
+def demo_bye(update, context):
+    text = "Thank you! Stay healthy."
+    update.message.reply_text(text=text, reply_markup=ReplyKeyboardRemove())
+    return ConversationHandler.END
+
+
+def demo_invalid_answer(update, context):
+    text = "That's not an answer from the given possibilities. Please use the buttons."
+    update.message.reply_text(text=text)
+
+
+def cancel(update, context):
+    text_cancel = "Bye! I hope we can talk again some day."
+    update.message.reply_text(text=text_cancel, reply_markup=ReplyKeyboardRemove())
+    return ConversationHandler.END
+
+
+demo_conv_handler = ConversationHandler(
+    entry_points=[CommandHandler("demo", demo_intro)],
+    states={
+        DemoDecisions.FAVPIC: [
+            MessageHandler(Filters.photo, demo_family),
+            MessageHandler(Filters.all, demo_family_invalid_input)
+        ],
+        DemoDecisions.FAMILY: [
+            MessageHandler(Filters.text & (filter_yes | filter_no), demo_amazon)
+        ],
+        DemoDecisions.AMAZON: [
+            MessageHandler(Filters.text & Filters.regex(r"[0-9]|10"), demo_durable)
+        ],
+        DemoDecisions.DURABLE: [
+            MessageHandler(Filters.text & Filters.regex(r"Mohammed Ali|Covid-19"), demo_difficult)
+        ],
+        DemoDecisions.DIFFICULT: [
+            MessageHandler(Filters.text & Filters.regex(r"Chinese Wall|Golden-Gate-Bridge|Eiffel Tower|Colosseum|Covid-19 vaccine"), demo_michael)
+        ],
+        DemoDecisions.MICHAEL: [
+            MessageHandler(Filters.text & Filters.regex(r"Yes|No|Not so sure"), demo_winner)
+        ],
+        DemoDecisions.WINNER: [
+            MessageHandler(Filters.text & (Filters.regex(r"") | Filters.regex(r"")), demo_bye)
```

Comment:
Either the regex is wrong here or I am.

```suggestion
            MessageHandler(Filters.text & Filters.regex(r"Covid-19|Humanity")), demo_bye)
```

Maybe we should also reply differently in demo_bye depending on their decision?

Reply:
Reply differently? Do you have a suggestion?